ui/components/multichain/multi-srp/srp-list/srp-list.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { KeyringMetadata, KeyringObject } from '@metamask/keyring-controller';
+import Card from '../../../ui/card';
+import {
+  Box,
+  IconName,
+  Icon,
+  Text,
+  IconSize,
+} from '../../../component-library';
+import {
+  JustifyContent,
+  Display,
+  TextColor,
+  FlexDirection,
+  AlignItems,
+  BlockSize,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import {
+  getMetaMaskAccounts,
+  getMetaMaskHdKeyrings,
+} from '../../../../selectors/selectors';
+import { InternalAccountWithBalance } from '../../../../selectors/selectors.types';
+import { SrpListItem } from './srp-list-item';
+
+type KeyringObjectWithMetadata = KeyringObject & { metadata: KeyringMetadata };
+
+export const SrpList = ({
+  onActionComplete,
+  hideShowAccounts,
+}: {
+  onActionComplete: (id: string) => void;
+  hideShowAccounts?: boolean;
+}) => {
+  const hdKeyrings: KeyringObjectWithMetadata[] = useSelector(
+    getMetaMaskHdKeyrings,
+  );
+  // This selector will return accounts with nonEVM balances as well.
+  const accountsWithBalances: Record<string, InternalAccountWithBalance> =
+    useSelector(getMetaMaskAccounts);
+
+  const showAccountsInitState = useMemo(
+    () =>
+      hdKeyrings.reduce(
+        (acc: Record<string, boolean>, _, index) => ({
+          ...acc,
+          [index]: Boolean(hideShowAccounts), // if hideShowAccounts is true, show all accounts by default
+        }),
+        {},
+      ),
+    [hdKeyrings, hideShowAccounts],
+  );
+
+  const [showAccounts, setShowAccounts] = useState<Record<string, boolean>>(
+    showAccountsInitState,
+  );
+
+  return (
+    <Box padding={4} data-testid="srp-list">
+      {hdKeyrings.map((keyring, index) => (
+        <Card
+          key={`srp-${index + 1}`}
+          data-testid={`hd-keyring-${keyring.metadata.id}`}
+          onClick={() => onActionComplete(keyring.metadata.id)}
+          className="select-srp__container"
+          marginBottom={3}
+        >
+          <Box
+            display={Display.Flex}
+            flexDirection={FlexDirection.Row}
+            alignItems={AlignItems.center}
+            justifyContent={JustifyContent.spaceBetween}
+            paddingLeft={4}
+          >
+            <Box>
+              <Text>{`Secret Phrase ${index + 1}`}</Text>
+              {!hideShowAccounts && (
+                <Text
+                  variant={TextVariant.bodySm}
+                  color={TextColor.primaryDefault}
+                  className="srp-list__show-accounts"
+                  onClick={(event: React.MouseEvent) => {
+                    event.stopPropagation();
+                    setShowAccounts({
+                      ...showAccounts,
+                      [index]: !showAccounts[index],
+                    });
```

Comment:
And if we use an array instead of an object, that would be:
```suggestion
                    showAccounts[index] = !showAccounts[index];
                    setShowAccounts(showAccounts);
```